### ISSUE TITLE
Explain how to compose widgets from the outside

### ISSUE THREAD

**Issue by Korijn** [zoofIO/flexx#180]
```python
class Foo(ui.Widget):
    @event.prop
    def scene(self, scene):
        return scene

    def init(self):
        if self.scene is None:
            self.scene = ui.Widget()

        with ui.VBox():
            self.slider = ui.Slider()
            self.contents = ui.Widget(children=[self.scene])
            with self.contents:
                self.label = ui.Label(text='Bar')

scene = ui.Widget()
with scene:
    ui.Label(text='Baz')
foo = Foo(scene=scene)
```

This pattern, which allows you to provide widgets from the outside during construction, works around the requirement of calling the constructor to attach to the widget hierarchy. It should probably be documented somewhere, as it gives some more insight into how you can compose complicated UIs.

**Comment by eudoxos** [comment]
Is this what I need to create a composed widget with slider & label (current value) and use it easily, the same as built-in flexx.ui classes?

**Comment by Korijn** [comment]
That can be simpler. In this issue the scene property is used to pass an existing widget into the Foo widget.

Your case could be:

```python
class LabeledSlider(ui.Widget):
    def init(self):
        self.slider = ui.Slider()
        self.label = ui.Label()
        self.value = self.slider.value

    @event.connect('slider.value')
    def on_slider(self, *events):
        self.label.text = self.slider.value
```

Maybe it can be even simpler... @almarklein will know :)

**Comment by almarklein** [comment]
Yeah, that's about it, you can probably ditch the `self.value = self.slider.value`, and maybe you'd wrap it in an HBox. 

There is also the low-level way of making a single custom widget that contains a slider dom element and div/span to display the value. Similar to how checkbox and radiobutton are implemented.

BTW: I'd love to revamp the slider widget to not use the standard HTML slider, but a custom implementation, because the standard slider looks and behaves different in each browser. This new slider should definitely (be able to) show the current value out of the box.
